drivers/sensor/si7006/si7006.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2019 Vaishali Pathak <vaishali@electronut.in>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "si7006.h"
+
+struct si7006_data {
+	struct device *i2c_dev;
+	float temperature;
+	float humidity;
+};
+
+/**
+ * @brief function to read a byte
+ *
+ * @return 0 for success
+ */
+
+static int read_bytes(struct device *i2c_dev, u8_t addr, u8_t *data,
```

Comment:
Can you not use `i2c_reg_read_byte`?

Reply:
`i2c_reg_read_byte` only supports 1 byte register addresses, so that can't be used.

Reply:
`i2c_write_read`?
`i2c_burst_read`?

Reply:
switched to using `i2c_burst_read`